src/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs

```diff
@@ -212,6 +223,40 @@ internal override MethodCallExpression Rewrite(Expression instance, IList<Expres
         }
     }
 
+    internal class MethodCallExpression0 : MethodCallExpression, IArgumentProvider
```

Comment:
Can we seal types like this?  Or do we not do that in this library? (Or maybe something somewhere derives from this?)